eng/testing/linker/SupportFiles/Directory.Build.props

```diff
@@ -1,4 +1,8 @@
 <Project>
+  <PropertyGroup>
+    <BeforeTargetFrameworkInferenceTargets>$(BUILD_REPOSITORY_LOCALPATH)/eng/BeforeTargetFrameworkInference.targets</BeforeTargetFrameworkInferenceTargets>
```

Comment:
I think the intention of these projects is that they are as close to a user's project as possible. Do we need to add our custom build logic to them?

Reply:
Yes because it imports the `System.Text.Json\src\System.Text.Json.csproj` and the NuGet chokes on the restore due to the TFMs 😢

Reply:
@benaadams do you mind elaborating here? I don't see the S.T.J trimming tests importing the System.Text.Json source project, and neither do the generated console apps for each test. Can you show us what error you are seeing without this line? As @eerhardt pointed out, the idea of the trimming tests is to try to keep out our build logic as much as possible so that the tests representan isolated project that is closer to what a customer app would look like.

Reply:
https://github.com/dotnet/runtime/pull/40877#issuecomment-676601051 When I add the browser type `$(NetCoreAppCurrent)-Browser` to `<TargetFrameworks>` then the linker tests for `TrimmingTests/Microsoft.Extensions.Logging.Console.TrimmingTests.proj` which imports `System.Text.Json` start failing with 
```
Errors
    NuGet.targets(128,5): Invalid restore input. Invalid target framework 'unsupported'. 
    Input files:  
        src/libraries/System.Text.Json/src/System.Text.Json.csproj.          
        trimmingTests/projects/Microsoft.Extensions.Logging.Console.TrimmingTests/AddConsoleFormatterTests/linux-x64/project.csproj               
        src/libraries/Microsoft.Extensions.Logging.Console/tests/TrimmingTests/Microsoft.Extensions.Logging.Console.TrimmingTests.proj
```

![image](https://user-images.githubusercontent.com/1142958/90805267-4df4e400-e313-11ea-9754-102686d5b016.png)

Reply:
The non-specificity of "unsupported" is display; I think is this NuGet issue https://github.com/NuGet/Home/issues/5413

Reply:
`Microsoft.Extensions.Logging.Console.csproj` brings in the `src` version

https://github.com/dotnet/runtime/blob/c039f3c0682b3c6f82fcab2573c6d81e6eec61ad/src/libraries/Microsoft.Extensions.Logging.Console/src/Microsoft.Extensions.Logging.Console.csproj#L48